python/compress-image/main.py

```diff
@@ -0,0 +1,131 @@
+""" Compress image function using Tinypng and Krakenio API """
+import base64
+import json
+import tinify
+import requests
+
+KRAKEN_API_ENDPOINT = "https://api.kraken.io/v1/upload"
+KRAKEN_USER_AGENT = "Mozilla/5.0 (Windows NT 6.1; Win64; x64)AppleWebKit/\
+                537.36(KHTML, like Gecko)Chrome/40.0.2214.85 Safari/537.36"
+
+
+def krakenio_impl(variables):
+    """
+    Implements image optimization using the Kraken.io API.
+    Input:
+        variables (dict): A dictionary containing the
+                          required variables for optimization.
+    Returns:
```

Comment:
The validate docstring includes raised exceptions. The tinypng and kraken docstring don't.